src/masternode-sync.cpp

```diff
@@ -340,19 +337,22 @@ void CMasternodeSync::Process()
                     } else {
                         GetNextAsset();
                     }
+                    g_connman->ReleaseNodeVector(vNodesCopy);
                     return;
                 }
 
                 if (RequestedMasternodeAttempt >= MASTERNODE_SYNC_THRESHOLD * 3) return;
```

Comment:
missing `ReleaseNodeVector`.
There's other couple instances below (line 378 and 414).